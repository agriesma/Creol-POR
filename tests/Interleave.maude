load interpreter
mod PROGRAM is
pr CREOL-INTERPRETER .
op init : -> Configuration [ctor] .
eq init =
< "Test" : Cl | Inh: noInh, Par: noAid, Att: "x" |-> null , "y" |-> null, Mtds: 
  < "init" : Mtdname | Param: noAid, Latt: noSubst, Code: "x" ::= int(0) ; "y" ::= int(0) ; return ( emp ) > *
  < "run" : Mtdname | Param: noAid, Latt: noSubst, Code: ( "x" ::= int(1) ; "y" ::= int(2) ||| "y" ::= int(1) ; "x" ::= int(2) ) ; return ( emp ) >, Ocnt: 0 >

.
endm
