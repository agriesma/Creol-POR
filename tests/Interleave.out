==========================================
reduce in PROGRAM : init .
result Class: 
< "Test" : Cl | Inh: noInh, Par: noAid, Att: "x" |-> null, "y" |-> null, Mtds:
  < "init" : Mtdname | Param: noAid, Latt: noSubst, Code: "x" ::= int(0) ; "y"
    ::= int(0) ; return(emp) > *
  < "run" : Mtdname | Param: noAid, Latt: noSubst, Code: ("x" ::= int(1) ; "y"
    ::= int(2) ||| "y" ::= int(1) ; "x" ::= int(2)) ; return(emp) >,
  Ocnt: 0 >
Bye.
